src/gui/shell.cpp

```diff
@@ -101,39 +102,61 @@ bool Shell::setGuiFont(const QString& fdesc, bool force)
 		return false;
 	}
 
-	QStringList attrs = fdesc.split(':');
-	if (attrs.size() < 1) {
-		m_nvim->api0()->vim_report_error("Invalid font");
-		return false;
+	bool setShellFontSuccess{ false };
+
+	if (fdesc == "*") {
+		bool fontDialogSuccess = false;
+		QFont font{ QFontDialog::getFont(&fontDialogSuccess, QWidget::font(), this, {},
+			QFontDialog::MonospacedFonts) };
+
+		// The font selection dialog was canceled by the user.
+		if (!fontDialogSuccess) {
+			return false;
+		}
+
+		setShellFontSuccess = setShellFont(font.family(), font.pointSize(), font.weight(),
+			font.italic(), force);
 	}
+	else {
+		QStringList attrs = fdesc.split(':');
+		if (attrs.size() < 1) {
+			m_nvim->api0()->vim_report_error("Invalid font");
+			return false;
+		}
 
-	qreal pointSize = font().pointSizeF();
-	int weight = -1;
-	bool italic = false;
-	foreach(QString attr, attrs) {
-		if (attr.size() >= 2 && attr[0] == 'h') {
-			bool ok = false;
-			qreal height = attr.mid(1).toFloat(&ok);
-			if (!ok) {
-				m_nvim->api0()->vim_report_error("Invalid font height");
-				return false;
+		qreal pointSize = font().pointSizeF();
+		int weight = -1;
+		bool italic = false;
+		for (const auto& attr : attrs) {
+			if (attr.size() >= 2 && attr[0] == "h") {
```

Comment:
I think small typo here, instead of "h" should be as previously 'h'.
I've tested this fix against https://github.com/equalsraf/neovim-qt/issues/585 issue on my Windows machine and it looks fine:)

Reply:
You are correct... Nice catch!

It still works, but the typing is wrong, better to have `'h'`